include/kernel.h

```diff
@@ -3278,8 +3279,8 @@ __syscall int k_condvar_wait(struct k_condvar *condvar, struct k_mutex *mutex,
 
 struct k_sem {
 	_wait_q_t wait_q;
-	uint32_t count;
-	uint32_t limit;
+	unsigned int count;
+	unsigned int limit;
```

Comment:
Just stupid question: Is size_t does not better here and should provide 32-/64-bit architecture coverage?

Reply:
In an ideal world, yeah. Or even just `uint32_t`.

The issue is that the current API's _functions_ all take/return `unsigned int`, and I don't want to change that for backwards compatibility purposes. So either `sizeof(uint32_t) == sizeof(unsigned int)`, in which case this doesn't change anything, `sizeof(uint32_t) < sizeof(unsigned int)`, in which case the previous implementation would silently break occasionally, or `sizeof(uint32_t) > sizeof(unsigned int)`, in which case the previous implementation was wasting space for values that it could never reach.

So for now I'm aligning on `unsigned int` throughout.

Reply:
Okay. I understand. Then it is really looks like good solution at this moment. Thank you.